inflate.c

```diff
@@ -1149,17 +1150,33 @@ int ZEXPORT inflate(z_stream *strm, int flush) {
                 }
                 if (copy > state->length)
                     copy = state->length;
+                if (copy > left)
+                    copy = left;
+                left -= copy;
+                state->length -= copy;
+                if (copy >= sizeof(long)) {
```

Comment:
I think this one would definitely benefit from `if (left >= sizeof(...))`... except that `left -= copy` needs to be moved to after the test (in fact since it can go after the copy, it can even go outside of the if/else block).